libraries/bsg_manycore_cuda.cpp

```diff
@@ -2053,9 +2061,23 @@ static int hb_mc_device_tiles_set_config_symbols (hb_mc_device_t *device,
 
 
 
-		// Set tile's tile group index  __bsg_tile_group_id_x/y symbol.
+		// Set tile's tile group index  __bsg_tile_group_id_x/y symbols.
+		// and tile's flat tile group id (tg_id_y * grid_dim_x + tg_id_x)
+		// Grid is a 2D array of tile groups representing an application
+		// with each element of grid being a tile group with X/Y coordinates
+		// inside the grid to distinguish it from other tile groups 
+		// __bsg_tile_group_id_x/y symbols define the X/Y coordinates of
+		// a tile's tile group relative to grid origin.
+		// The flat tile group id is calculated using the grid dimensions 
+		// and the tile group X/Y coordinates relative to grid origin
+		// __bsg_tile_group_id = __bsg_tile_group_id_y * __bsg_grid_dim_x
+		//                       + __bsg_tile_group_id_x 
+		// The flat tile group id is used in bsg_print_stat messages 
+		// to distinguish between bsg_print_stat statements of different 
+		// tile groups that run on a specific tile in sequence. 
```

Comment:
bsg_tile_group_id uniquely identifies each tile group in a grid. The flat tile group id is calculated using the grid dimensions and the tile group X/Y coordinates relative to grid origin as follows: 
__bsg_tile_group_id = __bsg_tile_group_id_y * __bsg_grid_dim_x + __bsg_tile_group_id_x 

bsg_tile_group_id is used in bsg_print_stat to distinguish between unique tile group invocations that have been executed on the same tile.

Reply:
"Executed on the same tile" still trips me up, because it does not eliminate the possibility that the tile groups are executing at the same time (Not possible, I know)